### PR TITLE
Fix stream updates for space list

### PR DESCRIPTION
**Pull request by cp-ishita-g**
- [x] fix new added space not updating
- [x] fix selected space when we delete selected space
- [x] fix new added space selection

Simulator Screen Recording - iPhone 15 Pro - 2024-09-04 at 17 43 25

### CODE REVIEW

**Comment by kaushiksaliya** [comment]
when we delete space from edit space then list is updating but not map some time it is updating new data with old space data.

File: app/lib/ui/flow/home/home_screen_viewmodel.dart [hunk 1]
```diff
@@ -96,13 +91,25 @@ class HomeViewNotifier extends StateNotifier<HomeViewState> {
...
+      final selectedSpaceIndex = sortedSpaces
+          .indexWhere((space) => space.space.id == currentSpaceId);
+      if (selectedSpaceIndex > -1) {
+        final selectedSpace = sortedSpaces.removeAt(selectedSpaceIndex);
+        sortedSpaces.insert(0, selectedSpace);
+        updateSelectedSpace(selectedSpace);
+        state = state.copyWith(selectedSpace: selectedSpace);
+      }
+    }
+    state = state.copyWith(spaceList: sortedSpaces);
```
Comment (kaushiksaliya) [inline app/lib/ui/flow/home/home_screen_viewmodel.dart:106]: remove this from here and update spaceList state with selectedSpace in last.

File: app/lib/ui/flow/home/components/home_top_bar.dart [hunk 2]
```diff
@@ -92,6 +92,7 @@ class _HomeTopBarState extends State<HomeTopBar> with TickerProviderStateMixin {
 
   @override
   Widget build(BuildContext context) {
+    selectedIndex = widget.spaces.indexWhere((space) => space.space.id == widget.selectedSpace?.space.id);
```
Comment (kaushiksaliya) [inline app/lib/ui/flow/home/components/home_top_bar.dart:95]: you already update selectedIndex above, so no need to update again with same value if you update with new index then it is fine but here you do same as you done above, remove this.

Comment (cp-ishita-g) [reply app/lib/ui/flow/home/components/home_top_bar.dart:95]: whenever space list is open and there any change in ui that late init will not update it so we need to update at build also

**Comment by cp-ishita-g** [comment]
> when we delete space from edit space then list is updating but not map some time it is updating new data with old space data.

i have tested it, it is working fine when we delete space, can you please delete app and than try again

**Comment by kaushiksaliya** [comment]
without doing anything, selected space changing. 

[Screencast from 05-09-24 05:11:40 PM IST.webm](https://github.com/user-attachments/assets/38ca3208-1895-4a48-9ac4-5e603c248d42)

**Comment by kaushiksaliya** [comment]
some time when we remove any space and go to the home screen then showing other space members inside the current selected space.

**Comment by kaushiksaliya** [comment]
Fix location toggle, when we enable or disable location from edit space screen then it is not reflect on home screen.

[Fix_location_toggle.webm](https://github.com/user-attachments/assets/f076e20e-040d-4ad4-9afc-ebf3114c1e56)

**Comment by cp-ishita-g** [comment]
> Collaborator

it is working fine you can check below recording
Simulator Screen Recording - iPhone 15 Pro - 2024-09-11 at 10 12 58

**Comment by cp-ishita-g** [comment]
> some time when we remove any space and go to the home screen then showing other space members inside the current selected space.

can you check at below recording that after deleting space member are getting updated properly according to selected space
Simulator Screen Recording - iPhone 15 Pro - 2024-09-11 at 10 21 04

**Review by kaushiksaliya — APPROVED** [review]

**Pull request merged**